plugins/qiskit_executor/routes.py

```diff
@@ -232,27 +234,16 @@ def post(self, arguments: CircuitParameters):
             db_id=db_task.id,
             _external=True,
         )
-        PluginState.set_value(
-            QiskitExecutor.instance.identifier,
-            f"authentication_href_{db_task.id}",
-            auth_href,
-        )
-        PluginState.set_value(
-            QiskitExecutor.instance.identifier,
-            f"authentication_ui_href_{db_task.id}",
-            auth_ui_href,
-        )
-        PluginState.set_value(
-            QiskitExecutor.instance.identifier,
-            f"backend_selection_href_{db_task.id}",
-            backend_href,
-        )
-        PluginState.set_value(
-            QiskitExecutor.instance.identifier,
-            f"backend_selection_ui_href_{db_task.id}",
-            backend_ui_href,
-            commit=True,
-        )
+
+        db_task.data["urls"] = {
+            "authentication": auth_href,
+            "authentication_ui": auth_ui_href,
+            "backend_selection": backend_href,
+            "backend_selection_ui": backend_ui_href,
+        }
+
+        db_task.save(commit=True)
```

Comment:
You can probably remove the save in line 214

Reply:
Unfortunately, the `db_task.id` is needed for the URL generation (lines 216-236), which then need to be added again to the task. I haven't figured out a way around the 2 `db_task.save(commit=True)` calls.

Reply:
To get a database ID without comitting use `DB.session.flush()`. The objects must be added to the session first (either with `db_task.save()` or `DB.session.add(db_task)`).

Reply:
Would it make sense to replace `db_task.save(commit=True)` by `db_task.save(); DB.session.flush()`?

Reply:
Yes. A commit commits the whole transaction. If an error occurs while trying to generate the URLs you do want to roll back all actions in the DB but if you have comitted before then this rollback does not happen automatically. In some cases comitting cannot be avoided (e.g. when a different process needs access to the DB record) but in this case we only need to commit at the end.